### ISSUE TITLE
Update stats API algorithms to use tree_position_t

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tskit#2796]
We should be able to substantially simplify the stats API algorithms by using the tsk_tree_position_t class.

This should be done before we generalise to windows that are not [0, L) #2782

**Comment by jeromekelleher** [comment]
There may be other low-hanging fruit lying around in terms of incremental algorithms, which we might combine in here, or log another issue. Part of closing this issue should be to check for places that we can use the tree_position_t logic, and either go ahead with the change or create a new issue to track.

**Comment by benjeffery** [comment]
Closing for inactivity and labelling "future", please re-open if you plan to work on this.
